### PR TITLE
Patrol mission

### PR DESCRIPTION
**Pull request by jginesclavero**
### System modes inference
**Please, accept first my previous PRs https://github.com/MROS-RobMoSys-ITP/Pilot-URJC/pull/48 https://github.com/MROS-RobMoSys-ITP/Pilot-URJC/pull/50**

This PR adds the patrolling behavior of the Pilot-URJC:
- I have added a new world and the map of the scenario.
image
- I have added a BT as the mission controller.
- I have replaced the previous BT nodes by NavigateToWp node.

### Future works
- Check that the README contains enough information to reproduce the full pilot.

### CODE REVIEW

File: laser_resender/src/laser_resender_node.cpp [hunk 1]
```diff
@@ -58,8 +60,13 @@ class LaserResender : public rclcpp_lifecycle::LifecycleNode
 
   CallbackReturnT on_deactivate(const rclcpp_lifecycle::State & state) 
   {
-    RCLCPP_INFO(get_logger(), "[%s] Deactivating from [%s] state...", get_name(), state.label().c_str());
-    return CallbackReturnT::SUCCESS;
+    if (all_zero_error_)
```
Comment (fmrico) [inline laser_resender/src/laser_resender_node.cpp:63]: Pass the tests to all this software. These lines would not pass linters

File: mros_contingencies_sim/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+)
+
+install(
+  DIRECTORY include/
+  DESTINATION include/
+)
+
+if(BUILD_TESTING)
+  find_package(ament_lint_auto REQUIRED)
+  ament_lint_auto_find_test_dependencies()
```
Comment (fmrico) [inline mros_contingencies_sim/CMakeLists.txt:117]: Could we add at any moment? Maybe in another PR

File: mros_contingencies_sim/src/battery_contingency_node.cpp [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    rclcpp::QoS(10),
+    std::bind(&BatteryContingency::amclCallback,
+    this,
+    _1));
+  battery_pub_ = create_publisher<std_msgs::msg::Float64>(
+    "/robot/battery",
+    rclcpp::QoS(10));
+
+  battery_level_ = 100.0;
+  battery_consumption_ = 2.0;
```
Comment (fmrico) [inline mros_contingencies_sim/src/battery_contingency_node.cpp:36]: Use constants for this magic numbers

File: mros_contingencies_sim/src/battery_contingency_node.cpp [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+}
+
+float BatteryContingency::calculateDistance(
+  float current_x, float current_y, float old_x, float old_y)
+{
+  return sqrt((pow(current_x - old_x, 2) + pow(current_y - old_y, 2)));
```
Comment (fmrico) [inline mros_contingencies_sim/src/battery_contingency_node.cpp:42]: avoid pow as I said the other PR

File: mros_contingencies_sim/src/battery_contingency_node.cpp [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+  else
+    setOldposition(msg->pose.pose);
+  
+  battery_level_ = battery_level_ - distance * battery_consumption_;
+  if (battery_level_ < 0.0)
+    battery_level_ = 0.0;
+  std_msgs::msg::Float64 battery_msg;
+  battery_msg.data = battery_level_;
+  battery_pub_->publish(battery_msg);
+  if (battery_level_ < 15.0)
```
Comment (fmrico) [inline mros_contingencies_sim/src/battery_contingency_node.cpp:76]: Magic number (I reviewed this code in the other PR)

**Review by fmrico — CHANGES_REQUESTED** [review]

**Comment by jginesclavero** [comment]
I have added linter and I have fixed all.

**Review by fmrico — APPROVED** [review]

**Pull request merged**